### PR TITLE
Nxos tests

### PR DESCRIPTION
**Pull request by itdependsnetworks**
## Description
Update NXOS tests structure.

This was becoming too much to do in one swoop, I actually tried a few times, and couldn't complete and had to start over. Going to go with a batch approach. Once I have the first one in, it should be easier to do a few files at a time. 

## Motivation and Context
Update NXOS tests structure

## Impact (If any)
Change in testing strategy, including nxos tests moving forward. 

## Screenshots:
N/A

## Checklist:

- [x] I have updated the changelog.
- [x] I have updated the documentation (If applicable).
- [x] I have added tests to cover my changes (If applicable).
- [x] All new and existing tests passed.
- [x] All new code passed compilation.

### CODE REVIEW

**Review by ThomasJRyan — APPROVED** [review]

**Comment by ThomasJRyan** [comment]
Thank you for this Ken

File: tests/ci_folder_parsing.py [hunk 1]
```diff
@@ -449,6 +449,228 @@ def test_empty(self, steps, local_class, operating_system, token=None):
         "ShowSdwanSystemStatus": True,
         "ShowSdwanSoftware": True,
     },
+    "nxos": {
+        "aci": {
+            "ShowPlatformInternalHalPolicyRedirdst": True,
+            "ShowServiceRedirInfoGroup": True,
+        },
+        "ShowMacAddressTableBase": True, # missing test
```
Comment (tahigash) [inline tests/ci_folder_parsing.py:457]: they are not migrated yet. and will be coming on new PR, correct?

**Review by tahigash — APPROVED** [review]
left 1 comment. but approved. please fix conflict and merge.

File: tests/ci_folder_parsing.py (hunk 1 above)
Comment (itdependsnetworks) [reply tests/ci_folder_parsing.py:457]: True for all but the ones that are missing tests already (e.g. I am not looking to build tests that do not exist yet)

Comment (tahigash) [reply tests/ci_folder_parsing.py:457]: thanks for the reply. yes, you don't need to build ones which don't have even old style unittest.

**Comment by itdependsnetworks** [comment]
@tahigash it seems that in #408 you deleted the tests/ci_folder_parsing.py (which presumably removes testing). It is unclear what the motivation is, but being that you have commit rights to the origin repo and know your initial intention of removing the file, I think it makes sense for you to make the updates.

**Comment by ThomasJRyan** [comment]
> 
> 
> @tahigash it seems that in #408 you deleted the tests/ci_folder_parsing.py (which presumably removes testing). It is unclear what the motivation is, but being that you have commit rights to the origin repo and know your initial intention of removing the file, I think it makes sense for you to make the updates.

`genieparser/test/ci_folder_parsing.py` was moved to `genieparser/src/genie/libs/parser/utils/unittests.py` and turned into an importable module. This was done for an internal team's purpose. Functionally it's still the same

**Comment by ThomasJRyan** [comment]
Alright, looks like I got it merged. Thank you again Ken

**Pull request closed without merging**